File: backend/internal/domain/term.go
package domain

// CurrentTerm คือปีการศึกษา + ภาคเรียนปัจจุบันของโรงเรียน (is_current)
type CurrentTerm struct {
	SemesterID   string
	AcademicYear int // ปีการศึกษา พ.ศ. เช่น 2568
	Term         int // ภาคเรียน 1 หรือ 2
}

File: frontend/src/shared/schemas/term.ts
import { z } from "zod";

// ปี/เทอมปัจจุบัน — ตรงกับ backend service.CurrentTermDTO
export const currentTermSchema = z.object({
  has_current: z.boolean(),
  semester_id: z.string(),
  academic_year: z.number(),
  term: z.number(),
});
export type CurrentTerm = z.infer<typeof currentTermSchema>;
